packages/next/src/server/app-render/create-hanging-input-abort-signal.ts

```diff
@@ -0,0 +1,30 @@
+import type { PrerenderStoreModern } from './work-unit-async-storage.external'
+
+/**
+ * In a prerender, we may end up with hanging Promises as inputs due them
+ * stalling on connection() or because they're loading dynamic data. In that
+ * case we need to abort the encoding of arguments since they'll never complete.
+ */
+export function createHangingInputAbortSignal(
```

Comment:
is there a better home for this. Not generally a fan of function-as-file though it is very prevalent in our codebase. Makes renamign functions super annoying b/c then you also have to rename the file too

Reply:
I agree. I had this originally in `packages/next/src/server/app-render/dynamic-rendering.ts`, but had to move it in c609fb3 because Turbopack didn't tree shake properly which triggered errors for `process.nextTick` usage in edge bundles.